app/src/main/java/com/simplecity/amp_library/ui/adapters/PagerAdapter.java

```diff
@@ -21,8 +22,22 @@
      *
      * @param fragmentManager The {@link FragmentManager} to use
      */
-    public PagerAdapter(FragmentManager fragmentManager) {
+    public PagerAdapter(FragmentManager fragmentManager, boolean refreshPager) {
         super(fragmentManager);
+
+        /*
+		 * Removing all displayed fragments to avoid the problem when changing the displayed tabs
+         */
+        if (refreshPager) {
+            List<Fragment> childFragments = fragmentManager.getFragments();
+            if (childFragments != null && !childFragments.isEmpty()) {
+                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+                for (Fragment childFragment : childFragments) {
+                    if (childFragment != null) fragmentTransaction.remove(childFragment);
+                }
+                fragmentTransaction.commitAllowingStateLoss();
+            }
+        }
```

Comment:
This could (probably should) be moved out to a dedicated method called `refreshPager()`.